public/app/features/alerting/unified/utils/notification-policies.ts

```diff
@@ -158,30 +158,31 @@ function findMatchingAlertGroups(
 function getInheritedProperties(
   parentRoute: Route,
   childRoute: Route,
-  propertiesParentInherited?: Partial<InhertitableProperties>
-) {
-  const propsFromParent: InhertitableProperties = pick(parentRoute, INHERITABLE_KEYS);
-  const inheritableProperties: InhertitableProperties = merge({}, propertiesParentInherited, propsFromParent);
+  propertiesParentInherited?: Partial<InheritableProperties>
+): Partial<InheritableProperties> {
+  const propsFromParent: InheritableProperties = pick(parentRoute, INHERITABLE_KEYS);
+  const inheritableProperties: InheritableProperties = {
+    ...propsFromParent,
+    ...propertiesParentInherited,
+  };
 
-  // TODO how to solve this TypeScript mystery?
   const inherited = reduce(
     inheritableProperties,
     (inheritedProperties: Partial<Route> = {}, parentValue, property) => {
-      const parentHasValue = parentValue !== undefined;
+      const parentHasValue = parentValue != null;
 
+      const inheritableValues = [undefined, '', null];
```

Comment:
should we include `undefined` in this list?

Reply:
Why do we need ts-ignore here?

Reply:
`undefined` is included here in the list of `inheritableValues` – did you mean another line?

Reply:
The `@ts-ignore` is here because in the reducer it infers `property: string` instead of `keyof Route` :(